app/controllers/serviceLearning/routes.py

```diff
@@ -22,20 +22,23 @@
 @serviceLearning_bp.route('/serviceLearning/courseManagement', methods = ['GET'])
 @serviceLearning_bp.route('/serviceLearning/courseManagement/<username>', methods = ['GET'])
 def serviceCourseManagement(username=None):
-    if g.current_user.isStudent:
-        abort(403)
-    if g.current_user.isCeltsAdmin or g.current_user.isFaculty:
-        setRedirectTarget("/serviceLearning/courseManagement")
+    try:
         user = User.get(User.username==username) if username else g.current_user
+    except DoesNotExist:
+        abort(404)
+
+    isRequestingForSelf = g.current_user == user 
+    if g.current_user.isCeltsAdmin or (g.current_user.isFaculty and isRequestingForSelf):
+        setRedirectTarget("/serviceLearning/courseManagement")
```

Comment:
Instead of always coming back to `/serviceLearning/courseManagement` in `setRedirectTarget`, we should come back to the original route they came in on (including the username if necessary). Check out the flask `request` object

Reply:
On it!